battle/utils.py

```diff
@@ -15,19 +15,20 @@ def _monitor_battle():
         while not memory.main.battle_active():
             pass
         fast = False
+        old_game_speed = memory.main.get_game_speed()
         while memory.main.battle_active():
             if fast and memory.main.auditory_dialog_playing():
                 memory.main.set_game_speed(0)
                 fast = False
             elif not fast:
                 memory.main.set_game_speed(2)
                 fast = True
-        memory.main.set_game_speed(0)
+        memory.main.set_game_speed(old_game_speed)
 
     def wrapper(*args, **kwargs):
         if game_vars.get_battle_speedup():
             logger.debug(f"Speeding battle up: {func.__name__}")
-            monitor = threading.Thread(target=_monitor_battle)
+            monitor = threading.Thread(target=_monitor_battle, daemon=True)
```

Comment:
This fixes the CTRL-C issue when you kill the TAS during combat when 4x speed is auto-enabled for combat. Also tested this locally.